src/KafkaW/Msg.h

```diff
@@ -2,20 +2,31 @@
 
 #include <cstdint>
 #include <cstdlib>
+#include <functional>
 
 namespace KafkaW {
 // Want to expose this typedef also for users of this namespace
 using uchar = unsigned char;
 
 class Msg {
 public:
+  Msg() = default;
+  Msg(std::uint8_t const *Pointer, size_t Size,
+      std::function<void()> DataDeleter)
+      : DataPointer(Pointer), DataSize(Size), OnDelete(DataDeleter) {}
```

Comment:
May as well avoid the copy here: `OnDelete(std::move(DataDeleter))`

Reply:
Right.